<div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom"><a href="https://www.mathjax.org">
    <img style="height:1.25em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" />
</a></span> </div>

<header style="background-color:#e7e2ee; color:#461d7c; line-height:2em; padding-left:1em; padding-right:1em; bottom:0; float:bottom; text-align:right; float:right; font-family:serif; font-size:small;">Problem 4-31 ⋮ Çengel, Y. A., and Boles, M. A. (2001). <i>Thermodynamics: An Engineering Approach.</i> Boston: McGraw-Hill.</header>  




### Given   
A rigid 10-L vessel initially contains a mixture of liquid water and vapor at 100°C with 12.3 percent quality. The mixture is then heated until its temperature is 150°C. 


### Find 
Calculate the heat transfer required for this process. 

### Get Organized

Every good home has a good diagram.

![figure, cb8, 4-31](figures/ch04_cen98179_p04031_lg.jpg)

#### These things you should notice right away:
1. Closed system ⋮ constant mass  
1. Rigid vessel ⋮ constant volume, no boundary work  
1. Constant mass + constant volume $\rightarrow$ constant specific volume  
1. We can resolve both states ⋮ $(T_1,x_1),\ (T_2,v_2=v_1)$
1. We can assume $\rm \Delta KE = \Delta PE \approx 0$


<div style="font-size:2000%; font-weight:900; line-height:1em;">Attack</div>

Apply the First Law, with subscripts 1 and 2 representing initial and final states, respectively:

\begin{array}{c}
  E_{in} &-& E_{out} &=& E_{sys} \\
  Q_{in} &&&=& U_2 - U_1 \\ {} 
\end{array}

\begin{gather*}
  Q_{in} = m\left( u_2 - u_1 \right) \\
  m = \frac{V}{v} \qquad u_1 = u \left( T_1, x_1 \right) \qquad u_2 = u \left( T_2, v_2 = v_1 \right)
\end{gather*}




In [43]:
using Unitful, DataFrames
pretty(val,dig) = round(typeof(val),val,digits=dig);

In [87]:
V = 10u"L"
x1 = 0.123

vf_100°C = 1.0435e-3u"m^3/kg"
vg_100°C = 1.673u"m^3/kg"
uf_100°C = 418.94u"kJ/kg"
ug_100°C = 2506.5u"kJ/kg"

vf_150°C = 1.0905e-3u"m^3/kg"
vg_150°C = 0.3928u"m^3/kg"
uf_150°C = 631.68u"kJ/kg"
ug_150°C = 2559.5u"kJ/kg"

v  = vf_100°C + x1 * (vg_100°C - vf_100°C)
u1 = uf_100°C + x1 * (ug_100°C - uf_100°C)

m = V / v |> u"kg"

x2 = (v - vf_150°C)/(vg_150°C - vf_150°C)
u2 = uf_150°C + x2 * (ug_150°C - uf_150°C)

Q_in = m * (u2 - u1)


DataFrame(T=[100u"°C",150u"°C"], x=[x1,x2], v=[v,v], u=[u1,u2]) |> println
println("\n\n","Q_in = ", pretty(Q_in,3))


2×4 DataFrame
│ Row │ T         │ x        │ v                  │ u                │
│     │ Quantity… │ Float64  │ Quantity…          │ Quantity…        │
├─────┼───────────┼──────────┼────────────────────┼──────────────────┤
│ 1   │ 100 °C    │ 0.123    │ 0.206694 m^3 kg^-1 │ 675.71 kJ kg^-1  │
│ 2   │ 150 °C    │ 0.524888 │ 0.206694 m^3 kg^-1 │ 1643.57 kJ kg^-1 │


Q_in = 46.826 kJ


<div class="btn-success">&nbsp;<div class="badge"></div><i class="fa fa-bomb"></i><strong> Boom</strong></div>
<div class="panel-success"><div class="panel-heading"><div class="badge"></div> 

\begin{equation*}
   Q_{in} = 46.826\,\text{kJ}
\end{equation*}  


</div></div>

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>